### ISSUE TITLE
Ties are handled incorrectly

### ISSUE THREAD

**Issue by MattiasBuelens** [egonny/advent-of-code-slack-notifier#1]
Today at 07:21:56, the top of our leaderboard was:

1. Kruptein (217 points)
2. Frederik Goovaerts (203 points)
3. Mattias Buelens (192 points)
4. francis duvivier (192 points)

At 07:39:11, Francis completed day 12 part 2. Since he was the first to do so and there were 11 people on the leaderboard, he scored 11 points. The leaderboard became:
1. Kruptein (217 points)
2. Frederik Goovaerts (203 points)
3. francis duvivier (203 points)
4. Mattias Buelens (192 points)

The bot posted the following message:
> **francis duvivier** has completed :star2: **Day 12** (2) at 07:39:11!
> :arrow_down: **4th** Mattias Buelens (192 points)

This message does not show that Francis climbed up to the third position. The message should have been:
> **francis duvivier** has completed :star2: **Day 12** (2) at 07:39:11!
> :arrow_up: **3rd** francis duvivier (203 points)
> :arrow_down: **4th** Mattias Buelens (192 points)

I think the logic for handling ties (added in 67b2381a029b7943dafe77392233fe381baf9785) is incorrect. I'll think about how this should be handled when I'm back home, I'll send you a PR if I come up with a solution. 😉

**Comment by egonny** [comment]
Yeah so this was broken initially, and tried to quickfix it but clearly not correctly.
The issue was that if you have a person X and Y, where at time t_1:
```
1. Y (60 points)
2. X (59 points)
```

if at some time t_2 X now gains one point, then initially you could think that the new leaderboard would be:
```
1. X (60 points)
1. Y (60 points)
```
where a person Z would then be third. Here we can see that both have the same position and are sorted by name alphabetically.

This is not the case however, for some reason whoever was higher initially retains the same place; the actual leaderboard would be:
 ```
1. Y (60 points)
2. X (60 points)
```
And this is clearly an issue for sorting people according to score, as it is now clear that in order to sort correctly, we need to either:
* keep in mind the previous order and make sure that in case of a draw, positions are retained.
* calculate positions purely based on the timestamps of each day rather than the local score.

Since option 1 is easier here, I'll go for option 1 :stuck_out_tongue:.

**Comment by MattiasBuelens** [comment]
I think the proper solution would be to use `last_star_ts` as a tie breaker when computing the positions. If two people have the same `local_score`, then the first person to have achieved that score should have the better position.

**Comment by egonny** [comment]
Oh right, I forgot that field existed (sure beats going through all goals to get each `get_star_ts`.)

You're correct, just adding `last_star_ts` in case of a tiebreaker SHOULD make sure it's always correct.

**Comment by MattiasBuelens** [comment]
Let's just hope that two people with the same score never complete a challenge *at the exact same second*. 😛

**Comment by egonny** [comment]
Fixed by PR #2.
